backend/bin/deal-observer-backend.js

```diff
@@ -40,21 +52,45 @@ const dealObserverLoop = async (makeRpcRequest, pgPool) => {
       Sentry.captureException(e)
     }
     const dt = Date.now() - start
-    console.log(`Loop "${LOOP_NAME}" took ${dt}ms`)
+    console.log(`Loop "${DEAL_OBSERVER_LOOP_NAME}" took ${dt}ms`)
+
+    if (INFLUXDB_TOKEN) {
+      recordTelemetry(`loop_${slug(DEAL_OBSERVER_LOOP_NAME, '_')}`, point => {
+        point.intField('interval_ms', DEAL_OBSERVER_LOOP_INTERVAL)
+        point.intField('duration_ms', dt)
+      })
+    }
+    if (dt < DEAL_OBSERVER_LOOP_INTERVAL) {
+      await timers.setTimeout(DEAL_OBSERVER_LOOP_INTERVAL - dt)
+    }
+  }
+}
+
+const dealSubmitterLoop = async (pgPool, sparkApiBaseURL, dealIngestionAccessToken) => {
```

Comment:
I am thinking about moving this to separate file but this will make deployment harder.

Reply:
@pyropy please take a look at the discussion in https://github.com/filecoin-station/deal-observer/pull/42 where we are discussing how to refactor the code to make it easier to add more loops (IPNI lookup, your submission to spark-api, etc.)

Reply:
I have reached out to @NikolasHaimerl and also left a [comment](https://github.com/CheckerNetwork/deal-observer/pull/42/files#r1933861920). Maye we can discuss it there?

Reply:
I'm going to keep old way of structuring loops until #42 gets merged. If #42 gets merged before I am going to pull changes and refactor my code accordingly.